File: Unidad 5 C#/Programa5U5.cs
using System;

namespace Programa5U5
{
    internal class Programa5U5
    {
        static void Main(string[] args)
        {
            Console.Write("Ingrese una frase para contar las palabras: ");
            string frase = Console.ReadLine();

            int cantidadPalabras = ContarPalabras(frase);
            Console.WriteLine("Número de palabras: " + cantidadPalabras);
        }

        static int ContarPalabras(string cadena)
        {
            if (string.IsNullOrEmpty(cadena))
            {
                return 0;
            }

            string[] palabras = cadena.Split(new char[] { ' ', '\t', '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries);
            return palabras.Length;
        }
    }
}

File: Unidad 5 Java/Programa5U5.java
package com.mycompany.programa5u5;
import java.util.Scanner;

public class Programa5U5 {

    public static void main(String[] args) {
  
 Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese una frase para contar las palabras: ");
        String frase = sc.nextLine();

        int cantidadPalabras = contarPalabras(frase);
        System.out.println("Número de palabras: " + cantidadPalabras);
    }

    static int contarPalabras(String cadena) {
        if (cadena == null || cadena.isEmpty()) {
            return 0;
        }

        String[] palabras = cadena.split("\\s+");
        return palabras.length;
    }
}